### PR TITLE
NEXT-10854 support custom cache folders

### PR DESCRIPTION
**Pull request by yannick80**
### 1. What does this change do, exactly?
It allows the usage of a custom Kernel that moves the cache folder to e.g. /tmp folder to speed up environments like vagrant.

### 2. Describe each step to reproduce the issue or behaviour.
If a custom kernel that extends. \Shopware\Core\Kernel is used overriding getCacheDir moving it to a folder like e.g. /tmp, ThemeDump Command and BundleConfig Dumper are not working correctly anymore, because the files created cannot be found by the other processes.

### 3. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ x] I have squashed any insignificant commits
- [ x] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x ] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+title: NEXT-10854
```
Comment (shyim) [inline changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md:2]:
```suggestion
title: Support custom cache folders
```

File: changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+issue: NEXT-10854
+author: Yann Karl
+author_email: y.karl@webweit.de 
+author_github: yannick80
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md:6]:
```suggestion
author_github: @yannick80
```

File: changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+---
+# Core
+*  Renamed `cacheDir` to `projectDir`, updated write path for `dump` in `src/Core/Framework/Plugin/BundleConfigDumper.php`
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md:9]: Please use `Added`, `Changed`, `Removed` or `Deprecated` as keywords here.

```suggestion
*  Changed property name `cacheDir` to `projectDir`, updated write path for `dump` in `src/Core/Framework/Plugin/BundleConfigDumper.php`
```

File: changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md:1]: No need to mention the ticket in the changelog filename

File: changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md (hunk 2 above)
Comment (yannick80) [reply changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md]: done, thanks for the hint

File: changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md (hunk 3 above)
Comment (yannick80) [reply changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md]: also done, good to know for the next pr

File: changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md (hunk 1 above)
Comment (yannick80) [reply changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md]: done, thanks for the hint

File: changelog/_unreleased/2020-09-17-support-custom-cache-folders.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+title: Support custom cache folders
+issue: NEXT-10854
+author: Yann Karl
+author_email: y.karl@webweit.de 
+author_github: @yannick80
+---
+# Core
+*  Changed property name `cacheDir` to `projectDir`, updated write path for `dump` in `src/Core/Framework/Plugin/BundleConfigDumper.php`
+*  Changed parameter from `%kernel.cache_dir%` to `%kernel.project_dir%` in service description for `BundleConfigDumper` in `src/Core/Framework/DependencyInjection/plugin.xml`
+
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-17-support-custom-cache-folders.md:11]:
```suggestion
```

Comment (yannick80) [reply changelog/_unreleased/2020-09-17-support-custom-cache-folders.md]: line removed, sorry for missing that one

File: changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md (hunk 4 above)
Comment (yannick80) [reply changelog/_unreleased/2020-09-17-next-10854-support-custom-cache-folders.md]: file renamed, will remind that for the next pr

**Review by shyim — APPROVED** [review]

**Review by PaddyS — APPROVED** [review]

**Pull request merged**